src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -13,13 +13,21 @@
 
     class SagaPersistenceBehavior : HandlingStageBehavior
     {
-        public ISagaPersister SagaPersister { get; set; }
+        public ISagaPersister SagaPersister { get; private set; }
 
-        public ICancelDeferredMessages TimeoutCancellation { get; set; }
+        public ICancelDeferredMessages TimeoutCancellation { get; private set; }
 
-        public MessageHandlerRegistry MessageHandlerRegistry { get; set; }
+        public MessageHandlerRegistry MessageHandlerRegistry { get; private set; }
 
-        public SagaMetaModel SagaMetaModel { get; set; }
+        public SagaMetadataCollection SagaMetadataCollection { get; private set; }
```

Comment:
Because this is available on the `ReadOnlySettings` should we pull this out of there instead of using property injection?